src/beachbot/manipulators/roarmm1.py

```
@@ -82,24 +107,36 @@ def refresh_robot_state(self):
                     if qs_changed:
                         with self._joint_changed:
                             self._joint_changed.notify_all()
+                else:
+                    # response from command!
+                    # print("response from servoboard:", data.decode())
+                    # One can do something, check fi command was successful or so ... 
+                    pass
+
             except Exception as ex:
                 print("Read error:" + strdata.decode())
                 print("Exception:", ex)
                 self.close_io()
 
     def set_max_torque(self, jointnr, maxvalue):
-        if jointnr>0 and jointnr<6 and maxvalue>=0 and maxvalue<=1000:
-            data = json.dumps(
-                {
-                    "T": 100,
-                    f"Q{jointnr}": maxvalue
-                }
-            )
+        if jointnr > 0 and jointnr < 6 and maxvalue >= 0 and maxvalue <= 1000:
+            data = json.dumps({"T": 100, f"Q{jointnr}": maxvalue})
             self.write_io(data)
         else:
-            raise ValueError("joint number range is 1 to 5 and value range is percent x10 (0-1000)")
+            raise ValueError(
+                "joint number range is 1 to 5 and value range is percent x10 (0-1000)"
+            )
 
     def set_joint_targets(self, qs):
+        pos = qs[-1]
+        if pos < 0:
+            pos = 0
+        if pos > 1:
+            pos = 1
```

I really don't like this approach. If the value is outside the acceptable bounds, you should raise an error, and tell the user to fix, not just hide it like this.

---

There is maybe noise or the robot gripper is pushed above its man/max level for open close, not sure but this is not necessarily an error I guess

---

The values could be any value, not just ones that fit your proposed scenario. If they are wildly wrong, we would never know with this approach. If you want to accept slightly higher or lower values, you can add specific checks for this (i.e. truncate if >1 and < 1.2 or something like that. Otherwise this just hides potential errors.